dhcp/src/lib.rs

```diff
@@ -50,7 +50,7 @@ pub const DST_PORT: u16 = 67;
 pub const SRC_PORT: u16 = 68;
 
 /// Duration in seconds to wait for the DHCP server to send a response.
-const TIMEOUT_SECS: u32 = 10;
+const TIMEOUT_SECS: u32 = 1;
```

Comment:
A lot of tests are hard coded to expect 10 seconds for `CallAfter` :sweat_smile: 

I can change this to an argument if you need different timeout values.

Reply:
I can make that sort of change later, too.

Would you expect it to not even look for a response for 11 seconds? That is what I am seeing.

Reply:
> Would you expect it to not even look for a response for 11 seconds? That is what I am seeing.

I have seen that occur directly after link up.

The W5500 will let you send packets without the link being up, `phycfgr` gets checked for link up in `w5500_dhcp::Client.process`, but it isn't perfectly reliable.  I would guess that it is sending the first DHCPDISCOVER while the link isn't up.

Reply:
I think there more going on. When I drop that timeout to 1s I get more reasonable behavior.

Reply:
Opened #153 to change that from a hard coded value.